src/Psl/Regex/Type/capture_groups.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Regex\Type;
+
+use Psl\Type\TypeInterface;
+
+use function Psl\Dict\from_keys;
+use function Psl\Dict\unique;
+use function Psl\Type\shape;
+use function Psl\Type\string;
+
+/**
+ * @param list<array-key> $groups
+ *
+ * @return TypeInterface<array<array-key, string>>
+ *
+ * @psalm-suppress MixedReturnTypeCoercion - Psalm uses track of the types: Trust the return type above.
```

Comment:
is this still neeeded?

Reply:
Yes, it keeps on complaining about this. Not sure why that is. The types on `from_keys` seem right.

Reply:
/** @psalm-trace $var **/ is your friend :D

Reply:
Just did: The data going into the shape function is correct: `array<array-key, Psl\Type\TypeInterface<string>>`
So I guess it's a fault somewhere in the existing shape psalm plugin?

Gonna log off for the weekend and try again next week

Reply:
The issue is int the plugin:
https://github.com/azjezz/psl/blob/1.5.x/integration/Psalm/EventHandler/ShapeFunctionReturnTypeProvider.php#L56

It is a `TArray` instead of a `TKeyedArray`, meaning it lost the keys in the array (which makes sense!)
In other words : The plugin can't iterator over the properties, since they are not known.

I think the suppress is valid here since the `RegexCaptureGroupsFunctionReturnTypeProvider` makes the keys known again, unless you have a beter solution?